### ISSUE TITLE
DVector<unsigned char>::resize - mem.is_locked() is true

### ISSUE THREAD

**Issue by Keetz** [godotengine/godot#7697]
Windows 8
Godot 2.1.1-stable & Godot 2.1.2-stable

Running the game gives following error
![godot_mem_error](https://cloud.githubusercontent.com/assets/6707330/22502069/b30faf38-e86b-11e6-9944-0d6939f15809.png)

Sometimes after a few minutes, sometimes after an hour.

I am not really sure how to reproduce this, as it seems pretty random when it occurs. I can also confirm that this isn't happening on Ubuntu 16.04 as it is my daily OS.

Some quotes from IRC that might be helpfull (thanks to kitkat_):

> <kitkat_> multithreading shenanigans
> <kitkat_> SSE2 maybe? (since windows 8 needs SSE2)

> <kitkat_> godot\core\dvector.h line 365
> <kitkat_> It means the program is trying to resize the vector while another thread has locked it

**Comment by Faless** [comment]
So, I managed to reproduce the bug both on windows an linux, the problem is in the HTTPClient ( https://github.com/godotengine/godot/blob/2.1/core/io/http_client.cpp#L580 ) and is NOT related to threads (actually, using threads in the requests does not trigger the bug).
The reason is that when not using threads the request is done in non-blocking mode, so it fetch partial data.
If the data fetched is not complete (would block) he tries to resize the DVector (which is locked by the Write).
I'll give it another look tomorrow, in the meantime, here is the backrace:
```
#0  DVector<unsigned char>::resize (this=0x7fffffffd4c0, p_size=3817) at core/dvector.h:365
#1  0x00000000018993cf in HTTPClient::read_response_body_chunk (this=0x380d810) at core/io/http_client.cpp:580
#2  0x0000000000e0bd9e in HTTPRequest::_update_connection (this=0x38105a0) at scene/main/http_request.cpp:403
#3  0x0000000000e0d0e8 in HTTPRequest::_notification (this=0x38105a0, p_what=17) at scene/main/http_request.cpp:466
#4  0x0000000000e0fb1e in HTTPRequest::_notificationv (this=0x38105a0, p_notification=17, p_reversed=false) at scene/main/http_request.h:39
#5  0x0000000001790726 in Object::notification (this=0x38105a0, p_notification=17, p_reversed=false) at core/object.cpp:908
#6  0x0000000000dd33d1 in SceneTree::_notify_group_pause (this=0x37f2fc0, p_group=..., p_notification=17) at scene/main/scene_main_loop.cpp:900
#7  0x0000000000dd1822 in SceneTree::idle (this=0x37f2fc0, p_time=0.0170409996) at scene/main/scene_main_loop.cpp:528
#8  0x000000000042f41b in Main::iteration () at main/main.cpp:1604
#9  0x0000000000410a0f in OS_X11::run (this=0x7fffffffdb00) at platform/x11/os_x11.cpp:1934
#10 0x0000000000408755 in main (argc=3, argv=0x7fffffffdff8) at platform/x11/godot_x11.cpp:41
```

**Comment by Hinsbart** [comment]
closing as fixed by #7749/#7750
